src/pyedb/dotnet/edb_core/cell/voltage_regulator.py

```diff
@@ -0,0 +1,148 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from pyedb.dotnet.edb_core.cell.layout_obj import Connectable
+from pyedb.dotnet.edb_core.edb_data.padstacks_data import EDBPadstackInstance
+
+
+class VoltageRegulator(Connectable):
+    """Class managing EDB voltage regulator."""
+
+    def __init__(self, pedb, edb_object=None):
+        super().__init__(pedb, edb_object)
+        self._positive_remote_sense_pin = None
+        self._negative_remote_sense_pin = None
+
+    @property
+    def component(self):
+        """Retrieve voltage regulator component"""
+        try:
+            ref_des = self._edb_object.GetComponent().GetName()
+            if not ref_des:
+                return False
+            return self._pedb.components.instances[ref_des]
+        except:
+            return False
+
+    @component.setter
+    def component(self, value):
+        if not isinstance(value, str):
+            self._pedb.logger.error("refdes name must be provided to set vrm component")
+            return
+        if value not in self._pedb.components.instances:
+            self._pedb.logger.error(f"component {value} not found in layout")
+            return
+        self._edb_object.SetGroup(self._pedb.components.instances[value].edbcomponent)
+
+    @property
+    def id(self):
+        """Retrieve voltage regulator ID."""
+        return self._edb_object.GetId()
+
+    @property
+    def load_regulator_current(self):
+        """Retrieve load regulator current value"""
+        return self._edb_object.GetLoadRegulationCurrent().ToDouble()
+
+    @load_regulator_current.setter
+    def load_regulator_current(self, value):
+        _value = self._pedb.edb_value(value)
+        self._edb_object.SetLoadRegulationCurrent(_value)
+
+    @property
+    def load_regulation_percent(self):
+        """Retrieve load regulation percent value."""
+        return self._edb_object.GetLoadRegulationPercent().ToDouble()
+
+    @load_regulation_percent.setter
+    def load_regulation_percent(self, value):
+        _value = self._edb_object.edb_value(value)
+        self._edb_object.SetLoadRegulationPercent(_value)
+
+    @property
+    def name(self):
+        """Retrieve voltage regulator name."""
+        return self._edb_object.GetName()
+
+    @name.setter
+    def name(self, value):
+        if isinstance(value, str):
+            self._edb_object.SetName(value)
+
+    @property
+    def negative_remote_sense_pin(self):
+        """Retrieve negative remote sense pin."""
+        return self._negative_remote_sense_pin
```

Comment:
why not use GetNegRemoteSensePin

Reply:
Because it is cashed in the variable by the setter. Like I mentionned it seems the padstackinstance id changed when it is assigned to the VRM. I could not find it again